crates/aranya-daemon/src/api.rs

```
@@ -393,6 +462,42 @@ impl DaemonApi for Api {
         todo!();
     }
 
+    #[instrument(skip(self))]
+    async fn encrypt_psk_seed_for_peer(
+        self,
+        _: context::Context,
+        team: api::TeamId,
+        peer_enc_pk: EncryptionPublicKey<CS>,
+    ) -> aranya_daemon_api::Result<WrappedSeed> {
+        let enc_pk = self.pk.lock().expect("poisoned").enc_pk.clone();
```

Can we instead use tokio's mutex? This should be `self.pk.lock().await`, and then we don't need to worry about panics

---

I guess we should decide on consistent which one to use and be consistent.

You're right about `tokio`'s mutex, it's nice that you don't have worry about handling `Result`s. The [docs](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use) seem to suggest using the blocking mutex unless you need to hold the lock across `await` points which we typically don't do.

Now that we're adding/improving benchmarks, we could see if there's a significant performance difference between using only non-blocking or blocking mutexes

---

This doesn't need to use tokio mutex because the lock period is very short and not held across await. The poisoning panic issue is orthogonal, we could return an error or use a non-poisoning mutex. Anyway, doesn't need to be handled on this PR.